wd_helper.au3

```diff
@@ -1503,32 +1503,35 @@ EndFunc   ;==>_WD_GetBrowserVersion
 ;                  Failure - "" and sets @error to one of the following values:
 ;                  - $_WD_ERROR_InvalidValue
 ;                  - $_WD_ERROR_NotFound
-;; Author ........: Danp2
-; Modified ......: Mlipok
+; Author ........: Danp2
+; Modified ......: mLipok
 ; Remarks .......:
 ; Related .......:
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
 Func _WD_GetBrowserPath($sBrowser)
 	Local Const $sFuncName = "_WD_GetBrowserPath"
+	Local Const $sRegKeyCommon = '\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\'
 	Local $iErr = $_WD_ERROR_Success
 	Local $sEXE, $sPath = ""
-	Local $cRegKey = '\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\'
 
 	Local $iIndex = _ArraySearch($_WD_SupportedBrowsers, $sBrowser, Default, Default, Default, Default, Default, $_WD_BROWSER_Name)
 	If @error Then
 		$iErr = $_WD_ERROR_InvalidValue
 	Else
 		$sEXE = $_WD_SupportedBrowsers[$iIndex][$_WD_BROWSER_ExeName]
-		$sPath = RegRead("HKLM" & $cRegKey & $sEXE, "")
 
-		If @error Then $sPath = RegRead("HKCU" & $cRegKey & $sEXE, "")
+		; check HKLM or in case of error HKCU
+		$sPath = RegRead("HKLM" & $sRegKeyCommon & $sEXE, "")
+		If @error Then $sPath = RegRead("HKCU" & $sRegKeyCommon & $sEXE, "")
 
+		; if both checks not found key in the registry
 		If @error Then
 			$iErr = $_WD_ERROR_NotFound
 		Else
 			$sPath = StringRegExpReplace($sPath, '["'']', '') ; String quotation marks
+			If StringInStr($sBrowser, 'opera') Then $sPath = StringReplace($sPath, 'Launcher.exe', $sEXE) ; Registry entries can contain "Launcher.exe" instead "opera.exe"
```

Comment:
I would prefer that this gets handled elsewhere. For example, this is what I changed in `SetupOpera` when testing with #218 --

```autoit
;	_WD_CapabilitiesAdd('binary', _WD_GetBrowserPath("opera"))
;	ConsoleWrite("! _WD_GetBrowserPath() > " & _WD_GetBrowserPath("opera") & @CRLF)

	Local $sPath = StringReplace(_WD_GetBrowserPath("opera"), 'launcher', 'opera')
	_WD_CapabilitiesAdd('binary', $sPath)
	ConsoleWrite("! _WD_GetBrowserPath() > " & $sPath  & @CRLF)
```
This way this new function can avoid browser specific coding. You could add something like this as a remark --

`Returned value may require further adjustment by calling routine (ex: opera)`

Reply:
It should be changed internally in `_WD_GetBrowserPath` as it is used also in `_WD_UpdateDriver`

so we can use:
```autoit
$sPath = StringReplace($sPath, 'Launcher.exe', $sEXE) ; Registry entries can contain "Launcher.exe" instead "opera.exe"
```

instead:
```autoit
If StringInStr($sBrowser, 'opera') Then $sPath = StringReplace($sPath, 'Launcher.exe', $sEXE) ; Registry entries can contain "Launcher.exe" instead "opera.exe"
```

Reply:
In my testing, this wasn't necessary. `_WD_GetBrowserVersion` calls `_WD_GetBrowserPath`, and the correct version number is returned when using `launcher.exe`.

Reply:
Going to sleep.
I will check it again probably tomorrow.

Reply:
But in my opinion launcher is not browser and for this reason `_WD_GetBrowserPath()` functions should return path to `opera.exe` and not to `launcher.exe`
If somebody will need it in a future then this is his own way to change opera.exe to launcher.exe.

Reply:
The current proposed code is ready to use - it works.

I see that you add new PR [Support alternative browser locations in _WD_UpdateDriver](https://github.com/Danp2/au3WebDriver/pull/226).
I also add new PR here [_WD_UpdateDriver - better error handling](https://github.com/Danp2/au3WebDriver/pull/227)

I would like to propose to commit current changes for this particular PR [added _WD_GetBrowserPath() with Opera path support](https://github.com/Danp2/au3WebDriver/pull/223) and if you want to change it later please creat an issue related to them.

When we combine all current PR in [opera branch](https://github.com/Danp2/au3WebDriver/tree/supportOpera) then we can back to this issue.